app/models/events/pull_request.rb

```diff
@@ -40,44 +40,76 @@ class PullRequest < ApplicationRecord
               :number,
               presence: true
     validates :draft,
-              :merged,
               :locked,
               inclusion: { in: [true, false] }
     validates :github_id, uniqueness: true
 
-    class << self
-      def resolve(payload)
-        action = payload[:action]
-        public_send(action, payload) if handleable?(action)
-      end
+    attr_accessor :payload
 
-      # Actions
+    def resolve
+      return unless handleable?
 
-      def merged(payload)
-        PullRequestJobs::Merged.perform_later(payload)
-      end
+      handle_action
+    end
 
-      def opened(payload)
-        PullRequestJobs::Opened.perform_later(payload)
-      end
+    private
 
-      def closed(payload)
-        PullRequestJobs::Closed.perform_later(payload)
-      end
+    def handle_action
+      send(payload[:action])
+    end
 
-      def review_requested(payload)
-        PullRequestJobs::ReviewRequested.perform_later(payload)
+    def find_or_create_user(user_data)
+      User.find_or_create_by!(github_id: user_data[:id]) do |user|
+        user.node_id = user_data[:node_id]
+        user.login = user_data[:login]
       end
+    end
 
-      def review_request_removed(payload)
-        PullRequestJobs::ReviewRequestRemoved.perform_later(payload)
+    def find_or_create_pull_request(payload)
+      pr_data = payload[:pull_request]
+      pr = Events::PullRequest.find_or_create_by!(github_id: pr_data[:id]) do |preq|
+        preq.node_id = pr_data[:node_id]
+        preq.number = pr_data[:number]
+        preq.state = pr_data[:state]
+        preq.locked = pr_data[:locked]
+        preq.draft = pr_data[:draft]
+        preq.title = pr_data[:title]
+        preq.body = pr_data[:body]
       end
+      pr.assign_attributes(payload: payload)
+      pr
+    end
 
-      private
+    def handleable?
+      ACTIONS.include?(payload[:action])
+    end
 
-      def handleable?(action)
-        ACTIONS.include?(action)
-      end
+    # Actions
+
+    def open
+      open!
+      save!(opened_at: Time.current)
+    end
+
+    def merged
+      merged!
+      save!(merged_at: Time.current)
+    end
```

Comment:
Is this for handling `merged` actions? Iirc there is no merged action, but rather a closed action with a merged param = true 
https://developer.github.com/v3/activity/events/types/#pullrequestevent

Reply:
Be sure to check whether or not merging changes the closed_at attr as well

Reply:
Yes! You are right, thanks